lib/main.c

```diff
@@ -259,23 +272,187 @@ int surjectionproof_verify(const unsigned char *proof_data, const size_t proof_l
   return ret;
 }
 
-int ec_seckey_negate(unsigned char *key) {
+int ec_seckey_negate(unsigned char *key)
+{
   secp256k1_context *ctx = secp256k1_context_create(SECP256K1_CONTEXT_ALL);
   int ret = secp256k1_ec_seckey_negate(ctx, key);
   secp256k1_context_destroy(ctx);
   return ret;
 }
 
-int ec_seckey_tweak_add(unsigned char *key, const unsigned char *tweak) {
+int ec_seckey_tweak_add(unsigned char *key, const unsigned char *tweak)
+{
   secp256k1_context *ctx = secp256k1_context_create(SECP256K1_CONTEXT_ALL);
   int ret = secp256k1_ec_seckey_tweak_add(ctx, key, tweak);
   secp256k1_context_destroy(ctx);
   return ret;
 }
 
-int ec_seckey_tweak_mul(unsigned char *key, const unsigned char *tweak) {
+int ec_seckey_tweak_mul(unsigned char *key, const unsigned char *tweak)
+{
   secp256k1_context *ctx = secp256k1_context_create(SECP256K1_CONTEXT_ALL);
   int ret = secp256k1_ec_seckey_tweak_mul(ctx, key, tweak);
   secp256k1_context_destroy(ctx);
   return ret;
 }
+
+int ec_is_valid_xonly_pubkey(const unsigned char *key)
+{
+  secp256k1_context *ctx = secp256k1_context_create(SECP256K1_CONTEXT_ALL);
+  secp256k1_xonly_pubkey pubkey;
+  int ret = secp256k1_xonly_pubkey_parse(ctx, &pubkey, key);
+  secp256k1_context_destroy(ctx);
+  return ret;
+}
+
+int ec_is_valid_ec_pubkey(const unsigned char *key, size_t key_len)
+{
+  secp256k1_context *ctx = secp256k1_context_create(SECP256K1_CONTEXT_ALL);
+  secp256k1_pubkey pubkey;
+  int ret = secp256k1_ec_pubkey_parse(ctx, &pubkey, key, key_len);
+  secp256k1_context_destroy(ctx);
+  return ret;
+}
+
+int ec_point_compress(unsigned char *output, size_t output_len, const unsigned char *point, size_t point_len)
+{
+  secp256k1_context *ctx = secp256k1_context_create(SECP256K1_CONTEXT_ALL);
+  secp256k1_pubkey pubkey;
+  int ret = secp256k1_ec_pubkey_parse(ctx, &pubkey, point, point_len);
+  if (ret == 1)
+  {
+    ret = secp256k1_ec_pubkey_serialize(ctx, output, &output_len, &pubkey, (output_len == 33) ? SECP256K1_EC_COMPRESSED : SECP256K1_EC_UNCOMPRESSED);
+  }
```

Comment:
i don't get this. The name of the function is `ec_point_compress` so we want it to be compressed.. so why making this check?
```suggestion
    ret = secp256k1_ec_pubkey_serialize(ctx, output, &output_len, &pubkey, SECP256K1_EC_COMPRESSED);
  }
```

Reply:
the name is not really meaningful I agree: in fact the function let's to compress **and** uncompress a point